kivy/properties.pyx

```diff
@@ -1996,34 +2003,38 @@ cdef class ColorProperty(Property):
         Color value will be dispatched when set through indexing or slicing,
         but when setting with slice you must ensure that slice has 4 components
         with float values between 0-1.
+        Assingning color name as value is now supported.
+        Value `None` is allowed as default value for property.
     '''
 
-    def __init__(self, defaultvalue=None, **kw):
-        defaultvalue = defaultvalue or [1, 1, 1, 1]
+    def __init__(self, defaultvalue=0, **kw):
+        defaultvalue = \
+            [1.0, 1.0, 1.0, 1.0] if defaultvalue == 0 else defaultvalue
         super(ColorProperty, self).__init__(defaultvalue, **kw)
 
     cdef convert(self, EventDispatcher obj, x):
         if x is None:
             return x
-        if isinstance(x, string_types):
-            return ObservableList(self, obj, self.parse_str(obj, x))
-
+        cdef object color = x
         try:
-            count = len(x)
-        except TypeError as e:
+            if isinstance(x, string_types):
+                color = self.parse_list(obj, self.parse_str(obj, x))
+            color = self.parse_list(obj, color)
+        except Exception as e:
```

Comment:
I don't think we should be catching so broadly as `Exception`. Specifically, `parse_str`, and more deeply `get_color_from_hex` don't raise any explicit exceptions. So we shouldn't try handling any implicit exceptions it may raise, even if we immediately re-raise, because generally if you don't know the reason for the exception it's best to not catch it and change the exception class.

So I would specifically only put the try/except around `parse_list` and only catch `ValueError`.

Reply:
If we have `some_color = ColorProperty('matham')`, then with `Exception` we get this traceback:
```
 Traceback (most recent call last):
   File "kivy\properties.pyx", line 2021, in kivy.properties.ColorProperty.convert
   File "kivy\properties.pyx", line 2032, in kivy.properties.ColorProperty.parse_str
   File "D:\Work\Kivy Framework\kivy\kivy\utils.py", line 121, in get_color_from_hex
     for x in split('([0-9a-f]{2})', s.lower()) if x != '']
   File "D:\Work\Kivy Framework\kivy\kivy\utils.py", line 121, in <listcomp>
     for x in split('([0-9a-f]{2})', s.lower()) if x != '']
 ValueError: invalid literal for int() with base 16: 'matham'
 
 The above exception was the direct cause of the following exception:
 
 Traceback (most recent call last):
   File "D:\Work\Kivy Framework\kivy\dev\color_app.py", line 43, in <module>
     ColorApp().run()
   File "kivy\_event.pyx", line 190, in kivy._event.EventDispatcher.__cinit__
   File "kivy\properties.pyx", line 448, in kivy.properties.Property.link
   File "kivy\properties.pyx", line 419, in kivy.properties.Property.init_storage
   File "kivy\properties.pyx", line 2024, in kivy.properties.ColorProperty.convert
 ValueError: ColorApp.some_color has an invalid format (got 'matham')
```

and with try/except only around `parse_list` with `ValueError` exception we get this:
```
 Traceback (most recent call last):
   File "D:\Work\Kivy Framework\kivy\dev\color_app.py", line 43, in <module>
     ColorApp().run()
   File "kivy\_event.pyx", line 190, in kivy._event.EventDispatcher.__cinit__
   File "kivy\properties.pyx", line 448, in kivy.properties.Property.link
   File "kivy\properties.pyx", line 419, in kivy.properties.Property.init_storage
   File "kivy\properties.pyx", line 2020, in kivy.properties.ColorProperty.convert
   File "kivy\properties.pyx", line 2032, in kivy.properties.ColorProperty.parse_str
   File "D:\Work\Kivy Framework\kivy\kivy\utils.py", line 121, in get_color_from_hex
     for x in split('([0-9a-f]{2})', s.lower()) if x != '']
   File "D:\Work\Kivy Framework\kivy\kivy\utils.py", line 121, in <listcomp>
     for x in split('([0-9a-f]{2})', s.lower()) if x != '']
 ValueError: invalid literal for int() with base 16: 'matham'
```

The former gives more clear information to user at the end of traceback, such as class name, property name and that format of assigned color is invalid which helps with debug.

I suggest just to replace `Exception` with `ValueError` and keep sting `parse_str` inside of try/except block.

Reply:
Actually, we should use `(ValueError, TypeError)` to cover case when new value is not `Sized`.